examples/pipeline_tacotron2/text/text_preprocessing.py

```diff
@@ -63,23 +64,103 @@
+    if phonemizer == "DeepPhonemizer":
+        from dp.phonemizer import Phonemizer
+        global _phonemizer
+        _other_symbols = ''.join(list(_special) + list(_punctuation))
+        _phone_symbols_re = r'(\[[A-Z]+?\]|' + '[' + _other_symbols + '])'  # [\[([A-Z]+?)\]|[-!'(),.:;? ]]
+
+        if _phonemizer is None:
+            # using a global variable so that we don't have to relode checkpoint
+            # everytime this function is called
+            _phonemizer = Phonemizer.from_checkpoint(checkpoint)
+
+        # Example:
+        # sent = "hello world!"
+        # '[HH][AH][L][OW] [W][ER][L][D]!'
+        sent = _phonemizer(sent, lang='en_us')
+
+        # ['[HH]', '[AH]', '[L]', '[OW]', ' ', '[W]', '[ER]', '[L]', '[D]', '!']
+        ret = re.findall(_phone_symbols_re, sent)
+
+        # ['HH', 'AH', 'L', 'OW', ' ', 'W', 'ER', 'L', 'D', '!']
+        ret = [r.replace("[", "").replace("]", "") for r in ret]
+
+        return ret
+    else:
+        raise ValueError(f"The `phonemizer` {phonemizer} is not supported. "
+                         "Supported `symbol_list` includes `'DeepPhonemizer'`.")
+
+
+def text_to_sequence(sent: str,
+                     symbol_list: Union[str, List[str]] = "english_characters",
+                     phonemizer: Optional[str] = "DeepPhonemizer",
+                     checkpoint: Optional[str] = "./en_us_cmudict_forward.pt",
+                     cmudict_root: Optional[str] = "./") -> List[int]:
     r'''Converts a string of text to a sequence of IDs corresponding to the symbols in the text.
 
-      Args:
+    Args:
         sent (str): The input sentence to convert to a sequence.
+        symbol_list (str or List of string, optional): When the input is a string, available options include
+            "english_characters" and "english_phonemes". When the input is a list of string, ``symbol_list`` will
+            directly be used as the symbol to encode. (Default: "english_characters")
+        phonemizer (str, optional): The phonemizer to use. Only used when ``symbol_list`` is "english_phonemes".
+            Available options include "DeepPhonemizer". (Default: "DeepPhonemizer")
+        checkpoint (str, optional): The path to the checkpoint of the phonemizer. Only used when ``symbol_list`` is
+            "english_phonemes". (Default: "./en_us_cmudict_forward.pt")
+        cmudict_root (str, optional): The path to the directory where the CMUDict dataset is found or downloaded.
+            Only used when ``symbol_list`` is "english_phonemes". (Default: "./")
 
-      Returns:
+    Returns:
         List of integers corresponding to the symbols in the sentence.
+
+    Examples:
+        >>> text_to_sequence("hello world!", "english_characters")
+        [19, 16, 23, 23, 26, 11, 34, 26, 29, 23, 15, 2]
+        >>> text_to_sequence("hello world!", "english_phonemes")
+        [54, 20, 65, 69, 11, 92, 44, 65, 38, 2]
     '''
+    if symbol_list == "english_phonemes":
```

Comment:
nit1: when writing a error message, rather than stating "what it should NOT be", explaining "what it should be" is more user-friendly.
For example: `threshold value cannot be 0.5` -> `thireshold value must be smaller than 0.4`.

nit2: You can also combine these all three checks with `any` but it is rather subjective which one is more readable.

```python
if any(param is None for param in [phonemizer, checkpoint, cmudict_root]):
    raise ValueError(
        "When `symbol_list` is 'english_phonemes', "
        "all of `phonemizer`, `checkpoint`, and `cmudict_root` must be provided.")
```

Reply:
Thank you for the suggestion.
I would prefer nit2 in this case.
It's addressed [here](https://github.com/pytorch/audio/pull/1668/files#diff-2b633e14bbda0eeaa9126220bbcc9782404e90d7cf7a0a4b2ab1be29dda27e56R143-R146).